torch/csrc/jit/script/sugared_value.cpp

```diff
@@ -158,38 +159,41 @@ std::vector<std::shared_ptr<SugaredValue>> SimpleValue::asTuple(
                          << " cannot be used as a tuple";
 }
 
+TypePtr InitializingClassValue::getExpectedTypeForAttr(
```

Comment:
Can you clarify the end game here for the sugared methods? I was not expecting SimpleValue to end up with a new virtual method `getExpectedTypeForAttr`. Right now SugaredValue is a pretty clean API of purely virtual stub methods, and this one makes it pretty confusing.

Reply:
This is purely to avoid code duplication (since most setAttr logic is shared between regular values and intializing values). If it's cleaner to leave it out I can